src/BenchmarkDotNet/Toolchains/MonoWasm/WasmGenerator.cs

```diff
@@ -0,0 +1,79 @@
+
+    {
+        private string RuntimePackPath;
+        private string WasmAppBuilderAssembly;
+        private string MainJS;
+
+        public WasmGenerator(string targetFrameworkMoniker,
+                             string cliPath,
+                             string packagesPath,
+                             string runtimeFrameworkVersion,
+                             string runtimePackPath,
+                             string wasmAppBuilderAssembly,
+                             string mainJS
+                             )
+            : base(targetFrameworkMoniker,
+                   cliPath,
+                   packagesPath,
+                   runtimeFrameworkVersion)
+        {
+            RuntimePackPath = runtimePackPath;
+            WasmAppBuilderAssembly = wasmAppBuilderAssembly;
+            MainJS = mainJS;
+
+        }
+
+        protected override void GenerateProject(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
+        {
+            BenchmarkCase benchmark = buildPartition.RepresentativeBenchmarkCase;
+            var projectFile = GetProjectFilePath(benchmark.Descriptor.Type, logger);
+            using (var file = new StreamReader(File.OpenRead(projectFile.FullName)))
+            {
+                var (customProperties, sdkName) = GetSettingsThatNeedsToBeCopied(file, projectFile);
+
+                string content = new StringBuilder(ResourceHelper.LoadTemplate("WasmCsProj.txt"))
+                   .Replace("$PLATFORM$", buildPartition.Platform.ToConfig())
+                   .Replace("$CODEFILENAME$", Path.GetFileName(artifactsPaths.ProgramCodePath))
+                   .Replace("$CSPROJPATH$", projectFile.FullName)
+                   .Replace("$TFM$", TargetFrameworkMoniker)
+                   .Replace("$PROGRAMNAME$", artifactsPaths.ProgramName)
+                   .Replace("$RUNTIMESETTINGS$", GetRuntimeSettings(benchmark.Job.Environment.Gc, buildPartition.Resolver))
+                   .Replace("$COPIEDSETTINGS$", customProperties)
+                   .Replace("$CONFIGURATIONNAME$", buildPartition.BuildConfiguration)
+                   .Replace("$SDKNAME$", sdkName)
+                   .Replace("$RUNTIMEPACKDIR$", RuntimePackPath)
+                   .Replace("$WASMAPPBUILDERASSEMBLY$", WasmAppBuilderAssembly)
+                   .Replace("$MAINJS$", MainJS)
+                   .ToString();
+
+                File.WriteAllText(artifactsPaths.ProjectFilePath, content);
+            }
+
+        }
+
+        protected override string GetBinariesDirectoryPath(string buildArtifactsDirectoryPath, string configuration)
+    => Path.Combine(buildArtifactsDirectoryPath, "bin/net5.0/browser-wasm/publish/output");
+
+        private static string GetPackagesDirectoryPath(bool useTempFolderForRestore, string packagesRestorePath)
+    => packagesRestorePath ?? (useTempFolderForRestore ? Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()) : null);
```

Comment:
it looks like copy-pasted from the `CoreRtGenerator`:

https://github.com/dotnet/BenchmarkDotNet/blob/3223c94a92050147a02482a3810860703f0c5171/src/BenchmarkDotNet/Toolchains/CoreRt/Generator.cs#L80-L85

is this method needed? I can't see any usages of it